### PR TITLE
expose more settings for for text plugin

### LINKED ISSUE #359: expose inline mode for text plugin
for plugins we sometimes we need the option to have formating (so a simple input is not enough), but not allow paragraphs.
(formulas in headings for example)

### PR DESCRIPTION
**Pull request by elbotho**
- this exposed the settings
- adds `noLinebreaks` option that just catches

closes #359

### CODE REVIEW

File: packages/plugins/text/src/controls/default.tsx [hunk 1]
```diff
@@ -37,41 +37,50 @@ import { ColoredTextIcon } from './colors'
 
 export function DefaultControls(props: SubControlProps) {
   const { editor, config, pluginClosure, plugins } = props
+
+  if (!plugins) return null
```
Comment (inyono) [inline packages/plugins/text/src/controls/default.tsx:41]: Why is this needed? `plugins` should always be defined according to the types.

**Review by inyono — COMMENTED** [review]
Thanks for the PR. Looks good at first glance.

File: packages/plugins/text/src/controls/index.tsx [hunk 2]
```diff
@@ -20,14 +20,7 @@ export interface ControlProps {
...
   pluginClosure: SlatePluginClosure
-  plugins: {
-    suggestions?: boolean
-    math?: boolean
-    headings?: boolean
-    lists?: boolean
-    colors?: boolean
-    code?: boolean
-  }
+  plugins: TextConfig['plugins']
```
Comment (elbotho) [inline packages/plugins/text/src/controls/index.tsx:23]: hier bin ich mir unsicher warum die types gedoppelt waren (aus meiner sicht)
ping @inyono

Comment (inyono) [reply packages/plugins/text/src/controls/index.tsx:23]: Ah, yeah, that's the problem. `TextConfig['plugins']` also might be undefined because of `plugins?: ...`. So if you want to share the type, you'd have to create a new type for that and use it in those two places (or alternatively do some weird TS voodoo to get rid of the undefined).

Comment (elbotho) [reply packages/plugins/text/src/controls/index.tsx:23]: so overall it make sense to share the types here? Than I'll create a new shared type.

**Comment by elbotho** [comment]
Could `markdownShortcuts` or `autoLink` have a problem with the link/richtext not being available or be used to create forbidden plugins?

**Comment by inyono** [comment]
Superseded by #370.
